pkg/apis/job.min.io/v1alpha1/types.go

```diff
@@ -0,0 +1,109 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:defaulter-gen=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Namespaced,shortName=miniojob,singular=miniojob
+// +kubebuilder:printcolumn:name="Tenant",type=string,JSONPath=`.spec.tenant.name`
+// +kubebuilder:printcolumn:name="Phase",type=string,JSONPath=`.spec.status.phase`
+
+// MinIOJob is a top-level type. A client is created for it
+type MinIOJob struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// *Required* +
+	//
+	// The root field for the MinIOJob object.
+	Spec MinIOJobSpec `json:"spec,omitempty"`
+
+	// Status provides details of the state of the MinIOJob steps
+	// +optional
+	Status MinIOJobStatus `json:"status,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// MinIOJobList is a top-level list type.
+type MinIOJobList struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []MinIOJob `json:"items"`
+}
+
+// MinIOJobSpec (`spec`) defines the configuration of a MinIOJob object. +
+type MinIOJobSpec struct {
+	// *Required* +
+	//
+	// Service Account name for the jobs to run
+	ServiceAccountName string `json:"serviceAccountName"`
+
+	// *Required* +
+	//
+	// TenantRef Reference for minio Tenant to eun the jobs against
+	TenantRef TenantRef `json:"tenant"`
+
+	// Execution order of the jobs, either `parallel` or `sequential`.
+	// Defaults to `parallel` if not provided.
+	// +optional
+	Execution string `json:"execution"`
+
+	// FailureStrategy is the forward plan in case of the failure of one or more MinioJob pods
+	// Either `stopOnFailure` or `continueOnFailure`, defaults to `continueOnFailure`.
+	// +optional
+	FailureStrategy string `json:"failureStrategy"`
+
+	// *Required* +
+	//
+	// Commands List of MinioClient commands
+	Commands []CommandSpec `json:"commands"`
+}
+
+// CommandSpec (`spec`) defines the configuration of a MinioClient Command.
+type CommandSpec struct {
+	// *Required* +
+	//
+	// Operation is the MinioClient Action
+	Operation string `json:"op"`
+
+	// Args Arguments to pass to the action
+	// +optional
+	Args []string `json:"args,omitempty"`
```

Comment:
Args should be a `map[string]interface{}` to allow for multiple types of `args`

Reply:
kubebuilder does not like this idea
```
Generating deepcopy funcs
F1122 16:07:27.601935     364 deepcopy.go:753] DeepCopy of "interface{}" is unsupported. Instead, use named interfaces with DeepCopy<named-interface> as one of the methods.
```

I am keeping the type as `map[string]string` for now, looking for a more dynamic structure that is supported by kubebuilder

Reply:
I see this `Outdated` meaning, code was changed. So I will resolve this conversation in order to proceed.